### DISCUSSION TITLE
Introduce custom breakpoint type in extension

### DISCUSSION THREAD

**Discussion by vaclavHala** [microsoft/vscode-discussions#1020] [Extension Development QnA] [answered]
I'd like to support setting of both software and hardware breakpoints (https://en.wikipedia.org/wiki/Breakpoint#Hardware) in my extension for embedded C development. The challenge with this is user may wish to set both sw and hw breakpoints in single file so "language" of the breakpoint can't be abused to differentiate the type. Also both sw and hw breakpints may be conditional so abusing the expression field to mark the breakpoint is problematic, not to mention the UX of such implementation is terrible. There needs to be some additional field on the definition of the breakpoint at the DAP level I think.

 I have found these tickets for custom breakpoints in extension https://github.com/microsoft/vscode/issues/42965 but it seems their conclusion was to introduce logpoint as hardwired type of breakpoint and no facility for adding custom breakpoint types was added.

Is there some trick for working with breakpoints I could use to add custom properties to breakpoint, both at the level of DAP communication and adding some convenient UI for user to interact with this custom property?

**Comment by isidorn** [comment, marked as answer]
I think this is a fair feature request - but I do not think it is currently planned.
fyi @connor4312

**Comment by connor4312** [comment]
Duplicates https://github.com/microsoft/debug-adapter-protocol/issues/454
